twiist/provider/provider.go

```diff
@@ -2,64 +2,156 @@ package provider
 
 import (
 	"context"
+	"time"
 
+	"github.com/tidepool-org/platform/oauth"
+
+	"github.com/lestrrat-go/jwx/v2/jwk"
+
+	"github.com/tidepool-org/platform/data"
+	dataClient "github.com/tidepool-org/platform/data/client"
+	dataDeduplicatorDeduplicator "github.com/tidepool-org/platform/data/deduplicator/deduplicator"
+	dataTypesUpload "github.com/tidepool-org/platform/data/types/upload"
+	"github.com/tidepool-org/platform/twiist"
+
+	"github.com/tidepool-org/platform/auth"
 	"github.com/tidepool-org/platform/config"
 	dataSource "github.com/tidepool-org/platform/data/source"
-	"github.com/tidepool-org/platform/dexcom/fetch"
 	"github.com/tidepool-org/platform/errors"
 	"github.com/tidepool-org/platform/log"
 	oauthProvider "github.com/tidepool-org/platform/oauth/provider"
 	"github.com/tidepool-org/platform/pointer"
-	"github.com/tidepool-org/platform/task"
 )
 
 const ProviderName = "twiist"
+const MetadataKeyExternalSubjectID = "externalSubjectID"
 
 type Provider struct {
 	*oauthProvider.Provider
+	dataClient       dataClient.Client
 	dataSourceClient dataSource.Client
-	taskClient       task.Client
 }
 
-func New(configReporter config.Reporter, dataSourceClient dataSource.Client, taskClient task.Client) (*Provider, error) {
+// Compile time check for making sure Provider{} is a valid oauth.Provider
+var _ oauth.Provider = &Provider{}
+
+func New(configReporter config.Reporter, dataClient dataClient.Client, dataSourceClient dataSource.Client, jwks jwk.Set) (*Provider, error) {
 	if configReporter == nil {
 		return nil, errors.New("config reporter is missing")
 	}
+	if dataClient == nil {
+		return nil, errors.New("data client is missing")
+	}
 	if dataSourceClient == nil {
 		return nil, errors.New("data source client is missing")
 	}
-	if taskClient == nil {
-		return nil, errors.New("task client is missing")
+	if jwks == nil {
+		return nil, errors.New("jwks is missing")
 	}
 
-	prvdr, err := oauthProvider.NewProvider(ProviderName, configReporter.WithScopes(ProviderName))
+	prvdr, err := oauthProvider.NewProvider(ProviderName, configReporter.WithScopes(ProviderName), jwks)
```

Comment:
Seems to me you could just push all of the jwks loading into the oauthProvider.Provider.

Reply:
Makes testing much harder

Reply:
Fair.